platform/security/core/security-core-impl/src/main/java/ddf/security/config/impl/ConfigurationSecurityLogger.java

```diff
@@ -37,29 +38,37 @@
 
   @Override
   public void configurationEvent(ConfigurationEvent event) {
-    try {
-      String type = getType(event);
-      Configuration[] configuration =
-          configurationAdmin.listConfigurations("(service.pid=" + event.getPid() + ')');
+    AccessController.doPrivileged(
```

Comment:
❓ Why do we need to elevate the security privileges here? And should we be doing something similar during migration export:import as we are also reading and writing configurations to config admin.

Reply:
You need permission to access the local subject, if one is available and this runs the block with all of the permissions available to the caller's protection domain.

Reply:
❓ How about the second question? Would that be something that needs to be done to the import and the export command to allow them to properly access all configurations when they are exporting and re-importing?

Reply:
It should not be necessary when running a command.

Reply:
The reason I am asking is because export/import will require additional privileges that shouldn't be granted to anything else and since that code will be called by the Karaf shell and the commands we register there, we would need to extend our privileges.